build/linux/debian/dependencies-generator.ts

```diff
@@ -0,0 +1,148 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { spawnSync } from 'child_process';
+import { constants, statSync } from 'fs';
+import path = require('path');
+import { additionalDeps, bundledDeps, referenceGeneratedDepsByArch } from './dep-lists';
+import { getSysroot } from './install-sysroot';
+import { ArchString } from './types';
+
+// A flag that can easily be toggled.
+// Make sure to compile the build directory after toggling the value.
+// If false, we warn about new dependencies if they show up
+// while running the Debian prepare package task for a release.
+// If true, we fail the build if there are new dependencies found during that task.
+// The reference dependencies, which one has to update when the new dependencies
+// are valid, are in dep-lists.ts
+const FAIL_BUILD_FOR_NEW_DEPENDENCIES: boolean = true;
+
+export function getDependencies(buildDir: string, applicationName: string, arch: ArchString): Promise<string[]> {
+	// Get the files for which we want to find dependencies.
+	const nativeModulesPath = path.join(buildDir, 'resources', 'app', 'node_modules.asar.unpacked');
+	const findResult = spawnSync('find', [nativeModulesPath, '-name', '*.node']);
+	if (findResult.status) {
+		console.error('Error finding files:');
+		console.error(findResult.stderr.toString());
+		return Promise.resolve([]);
+	}
+
+	const files = findResult.stdout.toString().trimEnd().split('\n');
+
+	const appPath = path.join(buildDir, applicationName);
+	files.push(appPath);
+
+	// Add chrome sandbox and crashpad handler.
+	files.push(path.join(buildDir, 'chrome-sandbox'));
+	files.push(path.join(buildDir, 'chrome_crashpad_handler'));
+
+	// Generate the dependencies.
+	const dependencies: Promise<Set<string>>[] = files.map((file) => calculatePackageDeps(file, arch, buildDir));
```

Comment:
It is currently a promise because the http request to get the sysroot makes it async. I can move out the sysroot call.
`getDependencies` for both packages has an `ArchString` type, but the type actually depends on the local `types.ts` file in each folder. I'd rather refactor out the common code in a separate PR.

Reply:
Can you open a tracking issue for this refactor, thanks!